web3/pm.py

```diff
@@ -1,27 +1,220 @@
+        given package name and version, if they are published to the currently set registry.
+        """
+        self._validate_set_registry()
+        _, _, release_uri = self.registry.get_release_data(name, version)
+        return self.get_package_from_uri(release_uri.rstrip(b'\x00'))
+
+    def _validate_set_registry(self):
+        try:
+            self.registry
+        except AttributeError:
+            raise PMError(
+                "web3.pm does not have a set registry. "
+                "Please set registry with either: "
+                "web3.pm.set_registry(address) or "
+                "web3.pm.deploy_and_set_registry()"
+            )
+        if not isinstance(self.registry, Registry):
+            raise PMError(
+                "web3.pm requires a Registry instance to be set as the "
+                "web3.pm.registry attribute."
+            )
+
+
+class Registry(Contract):
+    def __init__(self, address, w3):
+        # only works with v.py registry in ethpm/assets
+        # abi verification for compatibility?
+        manifest = json.loads((ASSETS_DIR / 'vyper_registry' / '1.0.0.json').read_text())
+        registry_package = Package(manifest, w3)
+        self.registry = registry_package.get_contract_instance(
+            "registry", to_canonical_address(address)
+        )
+        self.w3 = w3
+        self.address = address
+
+    @classmethod
+    def deploy_new_instance(cls, w3):
+        manifest = json.loads((ASSETS_DIR / 'vyper_registry' / '1.0.0.json').read_text())
+        registry_package = Package(manifest, w3)
+        registry_deployer = Deployer(registry_package)
+        deployed_registry_package = registry_deployer.deploy("registry")
+        registry_address = deployed_registry_package.deployments.get_instance("registry").address
+        return cls(registry_address, deployed_registry_package.w3)
+
+    def release(self, name, version, manifest_uri):
+        """
+        Returns tx_receipt from adding a new release.
+        """
+        tx_hash = self.registry.functions.release(name, version, manifest_uri).transact()
+        return self.w3.eth.waitForTransactionReceipt(tx_hash)
+
+    def get_release_data(self, name, version):
+        """
+        Returns (package_name, version, manifest_uri) associated with given name, version.
+        """
+        release_id = self.registry.functions.getReleaseId(name, version).call()
+        return self.registry.functions.getReleaseData(release_id).call()
+
+    @property
+    def owner(self):
+        """
+        Returns the owner address.
+        """
+        return self.registry.functions.owner().call()
+
+    @to_tuple
+    def get_all_package_ids(self):
+        """
+        Returns (package_name, release_count) for every package on registry.
+        """
+        package_count = self.registry.functions.packageCount().call()
+        for index in range(0, package_count, 4):
```

Comment:
using the term `offset` here instead of `index` will be more inline with the vyper implementation.